### DISCUSSION TITLE
add a view and an edit link

### DISCUSSION THREAD

**Discussion by erakastrati** [Budibase/budibase#2431] [Help] [answered]
Hi there,

I have a task within a table to make two icons, an eye and a pen. When eye is clicked, you can only view the client and you cannot edit it, and in the pen part you have the editable part, as below:
<img width="1348" alt="Screen Shot 2021-08-23 at 10 06 19 AM" src="https://user-images.githubusercontent.com/74623805/130412708-555f293a-e19b-408a-9e8a-c35c31ea1f9d.png">

I created a new screen named /clients/view where I made the form disabled and I added the /clients/view link to the eye icon. The problem now is that in each costumer I get the same view link, I mean the costumer is always the same(Abdyl in my case), and I don't know how to make for each costumer it's own view. 
<img width="1283" alt="Screen Shot 2021-08-23 at 10 15 59 AM" src="https://user-images.githubusercontent.com/74623805/130414494-112c3bc0-a1e3-4e45-9b34-2ec888f70899.png">

I think I need a costumer ID in my link, but I don't know how to make it happen. 
Any suggestions?

**Comment by joebudi** [comment, marked as answer]
Hey! When adding the link, you will need to bind to the client ID as you eluded to. To do that:

- [ ] Click define actions
- [ ] click add action
- [ ] click navigate to
- [ ] type your screen into the input - /clients/
- [ ] click the lightning bolt
- [ ] locate and click the 'clients table.Clients_id' option (this might not be the exact wording)
- [ ] Click save and save again

Give that a try and let me know if it worked.

Joe

**Reply by erakastrati** [reply]
It sends me to the edit part, I am a little confused.
Do you have time for a 2 minute meet maybe?

**Reply by joebudi** [reply]
Can the same user view, and edit?

**Reply by erakastrati** [reply]
yeah, the same user can view the client and also edit the client, the view part is disabled to click, thats why I created a new screen /clients/view but I don't know if it is the best way possible.

**Reply by joebudi** [reply]
May I ask what is the point in having a view only option if the same user can also edit? Why not have the one edit?

**Reply by erakastrati** [reply]
Because at the client table not all the columns ms are shown, some of them are hidden, while in the view part all the table columns are shown. My client wants it that way, I just have to do it so...

**Reply by joebudi** [reply]
Create two screens:
clients/edit/:id
clients/view/:id

Then link to /clients/edit/{{ clients table.Clients_id }}
And /clients/view/{{ clients table.Clients_id }}

**Reply by erakastrati** [reply]
What should I choose while creating the screen? Clients - Detail or create from scratch?

**Reply by erakastrati** [reply]
Thanks @joebudi !! It is working great now, thanks a lot for the help :)

**Reply by joebudi** [reply]
Awesome. Please mark this discussion as anwered.
